tests/integration/nm/ovs_test.py

```diff
@@ -149,13 +147,23 @@ def _attach_port_to_bridge(port_state):
     port_profile_name = nm.ovs.PORT_PROFILE_PREFIX + port_state[OB.PORT_NAME]
 
     _create_proxy_port(port_profile_name, port_state)
-    if port_state[OB.PORT_TYPE] == OBPortType.SYSTEM:
+    if _is_ethernet(port_state[OB.PORT_NAME]):
         _connect_interface(port_profile_name, port_state)
-    elif port_state[OB.PORT_TYPE] == OBPortType.INTERNAL:
+    else:
         iface_name = port_state[OB.PORT_NAME]
         _create_internal_interface(iface_name, master_name=port_profile_name)
 
 
+def _is_ethernet(iface_name):
+    dev = nm.device.get_device_by_name(iface_name)
+    if not dev:
+        return False
+    return dev.get_device_type() in (
+        nm.nmclient.NM.DeviceType.ETHERNET,
+        nm.nmclient.NM.DeviceType.VETH,  # simulates ethernet in tests
```

Comment:
Wouldn't it be more correct to specify the types that are not "system" interfaces, which in practice means they are OVS ones.

Non system ifaces: UNKNOWN, GENERIC, OVS_*

Reply:
I prefer to do positive checks that assuming that if something is not X it is Y. I can change it, but I think the current form is easier to understand.

Reply:
Currently the code assumes (i.e. depends) on which interface types are used in the test env, which seems wrong to me.

I am good with a positive check here, how about checking against ‘is_internal_interface’? This should drop the assumptions on the env.

Reply:
Makes sense. Should be fixed now.